src/main/java/flaxbeard/thaumicexploration/misc/brazier/SoulBrazierUtils.java

```diff
@@ -0,0 +1,212 @@
+		if (aPlayer == null) {
+			return null;
+		}
+		ArrayList<String> aWarpData = readPlayerWarpQueueDataFromFile();
+		for (String warpData : aWarpData) {
+			if (warpData != null && warpData.length() > 0 && warpData.contains(aPlayer.toString())) {
+				String[] playerData = warpData.split("@");
+				if (playerData.length == 6) {
+					int aQueuedWarp = Integer.parseInt(playerData[1]);
+					int aX = Integer.parseInt(playerData[2]);
+					int aY = Integer.parseInt(playerData[3]);
+					int aZ = Integer.parseInt(playerData[4]);
+					int aDimID = Integer.parseInt(playerData[5]);
+					return new SoulBrazierQueueData(aPlayer, aQueuedWarp, aX, aY, aZ, aDimID);
+				}
+			}
+		}
+		return null;
+	}
+
+	public static boolean removePlayerDataFromWarpQueue(UUID aPlayer, int aX, int aY, int aZ, int aDimID) {
+		if (aPlayer == null) {
+			return false;
+		}
+		ArrayList<String> aNewDataList = new ArrayList<String>();
+		ArrayList<String> aWarpData = readPlayerWarpQueueDataFromFile();
+		for (String warpData: aWarpData) {
+			if (warpData != null && warpData.length() > 0) {
+				if (!warpData.contains(aPlayer.toString())) {
+					aNewDataList.add(warpData);
+				}		
+				else {
+					if (!warpData.contains(aX+"@"+aY+"@"+aZ+"@"+aDimID)) {
+						aNewDataList.add(warpData);
+					}
+				}
+			}
+		}
+		if (!aNewDataList.isEmpty()) {
+			writePlayerWarpQueueDataToFile(aNewDataList);
+		}
+		return true;
+	}
+
+	private static void writePlayerWarpQueueDataToFile(ArrayList<String> aData) {
+		File warpFile = getPlayerWarpDataFile();		
+		sortWarpDataList(aData);
+		BufferedWriter writer;
+		try {
+			if (!warpFile.exists()) {
+				warpFile.createNewFile();
+			}			
+			writer = new BufferedWriter(new FileWriter(warpFile.getPath()));			
+			for (String aPlayerData : aData) {
+				writer.write(aPlayerData);	
+				writer.newLine();				
+			}			    
+			writer.close();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private static ArrayList<String> readPlayerWarpQueueDataFromFile() {
+		File warpFile = getPlayerWarpDataFile();		
+		BufferedReader reader;
+		ArrayList<String> aData = new ArrayList<String>();
+		try {
+			reader = new BufferedReader(new FileReader(warpFile.getPath()));			
+			String line;
+			while ((line = reader.readLine()) != null && line.length() > 0) {
+				aData.add(line);
+			}			
+			reader.close();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+		return aData;
+	}
+
+	private static void sortWarpDataList(ArrayList<String> aData) {
+		Collections.sort(aData);
+	}
+
+	private static File getSaveDirectory() {
+		return new File(Minecraft.getMinecraft().mcDataDir, "saves/"+ThaumicExploration.MODID+"/");
+	}
+
+	private static File getPlayerWarpDataFile() {
+		if (PlayerWarpDataFile != null && PlayerWarpDataFile.exists()) {
+			return PlayerWarpDataFile;
+		}
+		String dir = getSaveDirectory().getPath();
+		File warpFile = new File(dir+PlayerWarpDataFileName);
```

Comment:
Wouldn't that create a single warp data file for all saves in single player caase?

Reply:
To my knowledge this folder and data is in the save folder, so unique to each world.

Reply:
You mean `Minecraft.getMinecraft().mcDataDir` is actually a world folder? Then why do you append "saves/" to it in getSaveDirectory()?

Reply:
Because that’s what Google searching suggested I do. 🤷🏻‍♀️

You must remember I haven’t done this stuff in a while so I’m a tad rusty.

Reply:
I have no idea what is actually true here, but if you are not sure either, I think that should be verified then?

Reply:
This file field is final. It won't change upon switching world (say, go from SSP to creative). So it's definitely problematic. You'd need a onServerStarting or onServerAboutToStart hook to reset this field somehow. 

The convention is to save these per save data in the save directory of overworld, i.e. `server.getEntityWorld()..getSaveHandler().getWorldDirectory()`.